### PR TITLE
Add yarn workspaces support

### PR DESCRIPTION
**Pull request by magelle**
## What did you implement:
Support for yarn workspaces when including external packages.
There were issue with the yarn.lock wich isn't in the project folder but in the yarn workspace root folder.

Closes #XXXXX

## How did you implement it:
I added a dependency to get the root path of a yarn workspace. If the project is not a yarn workspaces project it default to the current behavior.

## How can we verify it:
I added a e2e test which represente the problem I had : examples/include-external-npm-packages-with-yarn-workspaces/

## Todos:

- [X] Write tests
- [x] Write documentation
- [X] Fix linting errors
- [X] Make sure code coverage hasn't dropped
- [X] Provide verification config / commands / resources
- [X] Enable "Allow edits from maintainers" for this PR
- [X] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

File: package.json
Comment (j0k3r) [inline package.json:36]: Why?

File: package.json
Comment (j0k3r) [inline package.json:45]: Not everyone is using yarn, why did you add it?

File: package.json
Comment (magelle) [reply package.json]: My mistake, I will remove it.
Miss the basic self review 🥲

File: package.json
Comment (magelle) [reply package.json]: same here

**Review by j0k3r — COMMENTED** [review]
Looks ok to me!

**Comment by j0k3r** [comment]
I'm waiting for a review from @vicary, in the mean time, I think you can squash our commits :)

**Review by vicary — APPROVED** [review]
I don't have a yarn workspace project to test this with, the changes looks fine.

Despite recent activities in GitHub, `find-yarn-workspace-root` has no new releases for 2 years. We should keep that in mind if yarn breaks this in 4+.

**Review by j0k3r — APPROVED** [review]

**Comment by gruckion** [comment]
Did you test if this works with Yarn PnP on Yarn Berry?

I was trying to package an external module whilst using Yarn workspaces and PnP. But I get warnings.

```
webpack compiled successfully
Warning: Could not check for peer dependencies of datadog-lambda-js. Set nodeModulesRelativeDir if node_modules is in different directory.
Package lock found - Using locked versions
Packing external modules: datadog-lambda-js@^6.87.0
Warning: Could not check for peer dependencies of datadog-lambda-js. Set nodeModulesRelativeDir if node_modules is in different directory.

✔ Service packaged (15s)
```

**Pull request merged**
